server/models.py

```diff
@@ -600,6 +605,18 @@ def _unflag_all(self, member_ids):
         if backup:
             backup.flagged = False
 
+    @transaction
+    def publish_grades(self, tag, hide=False):
+        """Publish student grades for the assignment."""
```

Comment:
Seems more like a toggle to me. 

I don't have a strong opinion on this - but it's worth thinking if this might be better as two methods?